closed/src/java.base/share/classes/sun/security/ec/NativeECDSASignature.java

```diff
@@ -584,8 +590,15 @@ protected boolean engineVerify(byte[] signature) throws SignatureException {
             return false;
         }
 
-        ECDSAOperations ops = ECDSAOperations.forParameters(params)
-                .orElseThrow(() -> new SignatureException("Curve not supported: " + params));
+        ECDSAOperations ops = null;
+        try{
+            ops = ECDSAOperations.forParameters(params)
+                    .orElseThrow(() -> new SignatureException("Curve not supported: " + params));
+        } catch (SignatureException se) {
+            if (!(ECUtil.isBrainpoolP512r1(params))) {
+                throw se;
+            }
+        }
```

Comment:
Again:
```java
        ECDSAOperations ops = ECDSAOperations.forParameters(params).orElse(null);
        if ((ops == null) && !NativeECUtil.isBrainpoolP512r1(params)) {
             throw new SignatureException("Curve not supported: " + params);
        }
```
Also, line 604 must anticipate that `ops` may be null:
```java
        if ((ops != null) && params.getCofactor() != 1) {
```

Reply:
Updated as suggested to avoid using exception handling for flow control. The null check for ops was also made.